python/ray/serve/_private/deployment_state.py

```diff
@@ -1398,6 +1398,21 @@ def _set_target_state_autoscaling(self, num_replicas: int) -> None:
         self._save_checkpoint_func(writeahead_checkpoints={self._id: target_state})
         self._target_state = target_state
 
+    @staticmethod
+    def get_capacity_adjusted_num_replicas(
+        num_replicas: int, target_capacity: Optional[float]
+    ) -> int:
+        """Return the target state `num_replicas` adjusted by the `target_capacity`.
+
+        The output will never be lower than `1`.
```

Comment:
> The output will never be lower than `1`.

What does this mean for scale-to-zero? Can deployments be scaled to zero if the config has a `target_capacity`?

Reply:
good catch. fixed and added test case for scale-to-zero.